### PR TITLE
Test/pro 4481/hsm

### PR DESCRIPTION
**Pull request by patrickvacek**
This greatly expands the HSM testing of oe-selftest and improves a bunch of other tests as well.

### CODE REVIEW

**Comment by OYTIS** [comment]
@patrickvacek Is it a backport?

**Comment by patrickvacek** [comment]
@OYTIS Nope, I wanted to get the test suite robust on pyro, since that's customer-facing and we want to be extra careful with any new changes to that, especially since the HSM support has been troublesome lately. I will also prepare a PR for rocko soon with all the recent HSM and testing changes, and at some point we can merge that into master as well.

**Comment by OYTIS** [comment]
@patrickvacek Okay, right, we do frontporting to master anyway.

**Comment by patrickvacek** [comment]
Yeah, normally I'd prefer to work directly on rocko (or even master after our recent discussion), but this was a special case where I thought it'd be best to start at the bottom.

**Comment by lbonn** [comment]
Here is an attempt to make the tests look a bit nicer (sample):
```
class SotaToolsTests(oeSelfTest):

    @classmethod
    def setUpClass(cls):
        logger = logging.getLogger("selftest")
        bb_vars = get_bb_vars(['SYSROOT_DESTDIR', 'base_prefix', 'libdir', 'bindir'],
                              'aktualizr-native')
        cls.sysroot = bb_vars['SYSROOT_DESTDIR'] + bb_vars['base_prefix']
        cls.sysrootlib = bb_vars['SYSROOT_DESTDIR'] + bb_vars['libdir']
        cls.sysrootbin = bb_vars['SYSROOT_DESTDIR'] + bb_vars['bindir']

        logger.info('Running bitbake to build aktualizr-native tools')
        bitbake('aktualizr-native')

    def runNativeCmd(self, cmd):
        env = dict(os.environ)
        env['LD_LIBRARY_PATH'] = self.sysrootlib
        return runCmd(cmd, env=env, native_sysroot=self.sysroot, ignore_status=True)

    def test_push_help(self):
        p = self.sysrootbin + "/garage-push"
        self.assertTrue(os.path.isfile(p), msg="No garage-push found (%s)" % p)
        result = self.runNativeCmd('garage-push --help')
        self.assertEqual(result.status, 0, "Status not equal to 0. output: %s" % result.output)

    def test_deploy_help(self):
        p = self.sysrootbin + "/garage-deploy"
        self.assertTrue(os.path.isfile(p), msg="No garage-deploy found (%s)" % p)
        result = self.runNativeCmd('garage-deploy --help')
        self.assertEqual(result.status, 0, "Status not equal to 0. output: %s" % result.output)

    def test_garagesign_help(self):
        p = self.sysrootbin + "/garage-sign"
        self.assertTrue(os.path.isfile(p), msg="No garage-sign found (%s)" % p)
        result = self.runNativeCmd('garage-sign --help')
        self.assertEqual(result.status, 0, "Status not equal to 0. output: %s" % result.output)
```

@patrickvacek If you're fine with it, I can make the changes (in qemu tests as well) and push on your branch

**Comment by patrickvacek** [comment]
@lbonn Yeah, looks good! I'm still learning my way around all the oe-selftest functions and tools, so any general improvements are welcome.

**Comment by patrickvacek** [comment]
@lbonn I'm getting errors like this from four tests now:
```
======================================================================
FAIL: test_cert_provider_help (oeqa.selftest.updater.AktualizrToolsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/patrick/Code/updater_qemu_pyro/meta-updater/lib/oeqa/selftest/updater.py", line 138, in test_cert_provider_help
    self.runNativeCmd('aktualizr_cert_provider --help')
  File "/home/patrick/Code/updater_qemu_pyro/meta-updater/lib/oeqa/selftest/updater.py", line 132, in runNativeCmd
    self.assertEqual(result.status, 0, "Status not equal to 0. output: %s" % result.output)
AssertionError: 127 != 0 : Status not equal to 0. output: aktualizr_cert_provider: error while loading shared libraries: libcrypto.so.1.0.2: cannot open shared object file: No such file or directory
```
This was the same error that prompted me to force in the LD_LIBRARY_PATH bit. I'm not sure why your refactor isn't working right.

**Comment by lbonn** [comment]
Hum, I indeed made a mistake while translating your code and the problem does not manifest on my machine...

Is it working now?

**Comment by patrickvacek** [comment]
@lbonn Thanks! I was seeing a minor logging issue, so I made a quick fix for that. Otherwise, I think it looks good.

@EpicNuts The only "real" code changes are bumping the aktualizr version by a bit and changing `cert_provider` so that it doesn't get deleted anymore. That's in anticipation of some work that Anton is doing. It effectively changes nothing at the moment, but it is necessary for some of the oe-selftest cases to pass.

I think this is ready to go now, so if you do an end-to-end test and oe-selftest and it all looks good, please merge!

File: lib/oeqa/selftest/updater.py [hunk 1]
```diff
@@ -174,19 +218,164 @@ def tearDownLocal(self):
...
+        stdout, stderr, retcode = self.qemu_command(softhsm2_command)
+        self.assertNotEqual(retcode, 0, 'softhsm2-tool succeeded before initialization: ' +
+                        stdout.decode() + stderr.decode())
+
+        # Run cert_provider.
+        bb_vars = get_bb_vars(['SYSROOT_DESTDIR', 'bindir', 'libdir',
+                               'SOTA_PACKED_CREDENTIALS'], 'aktualizr-native')
+        creds = bb_vars['SOTA_PACKED_CREDENTIALS']
+        bb_vars_prov = get_bb_vars(['STAGING_DIR_NATIVE', 'libdir'], 'aktualizr-hsm-prov')
+        config = bb_vars_prov['STAGING_DIR_NATIVE'] + bb_vars_prov['libdir'] + '/sota/sota_implicit_prov.toml'
```
Comment (lbonn) [inline lib/oeqa/selftest/updater.py]: Should be `/sota/sota_hsm_prov.toml`

https://github.com/advancedtelematic/meta-updater/pull/261#discussion_r169684947

**Comment by EpicNuts** [comment]
Tests passed. Merging in.

**Pull request merged**